python/pyarrow/compute.py

```diff
@@ -70,6 +70,7 @@
     function_registry,
     get_function,
     list_functions,
+    _group_by
```

Comment:
Is there a reaons for exposing this publicly? Is this just a leftover from previous attempts?

Reply:
It's to make it available when `_pc()` is used to access compute functions from other modules/files (in this case from `table.pxi`). I adhered to that practice instead of injecting a `import pyarrow._compute`